includes/definitions/discovery/esphome.yaml

```diff
@@ -0,0 +1,79 @@
+mib_dir: esphome
+modules:
+    os:
+        hardware:
+            - ESPHOME-ESP8266-SNMP-MIB::chipModel.0
+            - ESPHOME-ESP8266-SNMP-MIB::chipRevision.0
+        hardware_template: 'Model: {{ $ESPHOME-ESP8266-SNMP-MIB::chipModel.0 }} Revision: {{ $ESPHOME-ESP8266-SNMP-MIB::chipRevision.0 }}'
+        version: SNMPv2-MIB::sysDescr.0
+        version_regex: '/(?<version>version [0-9]+\.[0-9]+\.[0-9]+)/'
+    mempools:
+        data:
+            - # Heap ESP32 only
+                oid: ESPHOME-ESP32-SNMP-MIB::heap
+                total: ESPHOME-ESP32-SNMP-MIB::heapSize
+                free: ESPHOME-ESP32-SNMP-MIB::freeHeap
+                precision: 1
+                descr: 'Heap'
+                class: Heap
+                skip_values:
+                    - # Skip on esp8266 as its unsupported
+                        oid: ESPHOME-ESP8266-SNMP-MIB::chipType
+                        op: '='
+                        value: 8266
```

Comment:
I dont think you need to skip like this, if it doesnt cause any issues

Reply:
> I dont think you need to skip like this, if it doesnt cause any issues

The 8266 variant supports the `free:` but not the `total:`. I can take out the `skip_values:` if `mempools:` will not care about finding incomplete data on the 8266 variant.